policy/release/rpm_repos.rego

```diff
@@ -111,12 +111,26 @@ all_purls_with_repo_ids contains purl_obj if {
 }
 
 all_rpm_purls contains purl if {
-	some sbom in _all_sboms
-	some component in sbom.components
-	purl := component.purl
+	some sbom in lib.sbom.all_sboms
+	some purl in _purls(sbom)
 	_is_rpmish(purl)
 }
 
+_purls(sbom) := purls if {
+	cyclonedx_purls := {component.purl |
+		some component in sbom.components
+	}
+
+	spdx_purls := {ref.referenceLocator |
+		some pkg in sbom.packages
+		some ref in pkg.externalRefs
+		ref.referenceType == "purl"
+		ref.referenceCategory == "PACKAGE-MANAGER"
+	}
+
+	purls := cyclonedx_purls | spdx_purls
+}
+
```

Comment:
Nitpick: perhaps splitting these up might be nicer:

```suggestion
_cyclonedx_purls(sbom) := {component.purl |
	some component in sbom.components
}

_spdx_purls(sbom) := {ref.referenceLocator |
	some pkg in sbom.packages
	some ref in pkg.externalRefs
	ref.referenceType == "purl"
	ref.referenceCategory == "PACKAGE-MANAGER"
}

_purls(sbom) := _cyclonedx_purls(sbom) | _spdx_purls(sbom)
```

Reply:
Good idea. Done!